babel.config.js

```diff
@@ -103,6 +103,10 @@ module.exports = function getBabelConfig(api) {
     ]);
   }
 
+  if (process.env.MUI_ADD_IMPORT_EXTENSIONS === 'true') {
```

Comment:
Is there a reason to not always add the extensions?

Reply:
The `regressions` and `rollup` (umd) builds use this config file as well, and those break if we add the extensions.

When we remove the umd build, I could test if the issue with the `regressions` build is fixable, and we can remove this check. Does that make sense?

Reply:
Ideally al tests use builds that are as close to the real build as possible

Reply:
That makes sense, I'll look into it 👍🏼

Reply:
Hey @Janpot, I investigated the issue a bit. The `regressions` build uses `babel.config.js` to compile the packages as well as other files, like the docs data files and the regressions tests files. It also points to the `src` of the packages instead of the `build` (see the [regressions webpack config](https://github.com/mui/material-ui/blob/next/test/regressions/webpack.config.js#L56) and the [base webpack config](https://github.com/mui/material-ui/blob/next/webpackBaseConfig.js#L11)).

I was able to use the packages builds instead of the `src` code by doing

```diff
diff --git a/test/regressions/webpack.config.js b/test/regressions/webpack.config.js
index b4472eedd1..c9d02bb705 100644
--- a/test/regressions/webpack.config.js
+++ b/test/regressions/webpack.config.js
@@ -64,6 +64,20 @@ module.exports = {
       // Exclude polyfill and treat 'zlib' as an empty module since it is not required. next -> gzip-size relies on it.
       zlib: false,
     },
+    alias: {
+      ...webpackBaseConfig.resolve.alias,
+      '@mui/material': path.resolve(__dirname, '../../packages/mui-material/build'),
+      '@mui/icons-material': path.resolve(__dirname, '../../packages/mui-icons-material/build/esm'),
+      '@mui/lab': path.resolve(__dirname, '../../packages/mui-lab/build'),
+      '@mui/styled-engine': path.resolve(__dirname, '../../packages/mui-styled-engine/build'),
+      '@mui/styled-engine-sc': path.resolve(__dirname, '../../packages/mui-styled-engine-sc/build'),
+      '@mui/styles': path.resolve(__dirname, '../../packages/mui-styles/build'),
+      '@mui/system': path.resolve(__dirname, '../../packages/mui-system/build'),
+      '@mui/private-theming': path.resolve(__dirname, '../../packages/mui-private-theming/build'),
+      '@mui/base': path.resolve(__dirname, '../../packages/mui-base/build'),
+      '@mui/utils': path.resolve(__dirname, '../../packages/mui-utils/build'),
+      '@mui/joy': path.resolve(__dirname, '../../packages/mui-joy/build'),
+    },
   },
```

But even after doing that I still had to filter out some files and not add the file extensions to them:

```diff
diff --git a/babel.config.js b/babel.config.js
index 12d09a03c2..2d3ba4a891 100644
--- a/babel.config.js
+++ b/babel.config.js
@@ -103,10 +103,6 @@ module.exports = function getBabelConfig(api) {
     ]);
   }
 
-  if (process.env.MUI_ADD_IMPORT_EXTENSIONS === 'true') {
-    plugins.push(['babel-plugin-add-import-extension', { extension: useESModules ? 'mjs' : 'js' }]);
-  }
-
   return {
     assumptions: {
       noDocumentAll: true,
@@ -119,6 +115,12 @@ module.exports = function getBabelConfig(api) {
         exclude: /\.test\.(js|ts|tsx)$/,
         plugins: ['@babel/plugin-transform-react-constant-elements'],
       },
+      {
+        exclude: /(\/test\/regressions|packages\/mui-docs|docs)/,
+        plugins: [
+          ['babel-plugin-add-import-extension', { extension: useESModules ? 'mjs' : 'js' }],
+        ],
+      },
     ],
     env: {
       coverage: {
```

This is because the files in those folders do not have extensions, as we're not adding them. We could add them, but I think this is outside of this PR's scope.

In conclusion, I would do 

1. Maintain the `MUI_ADD_IMPORT_EXTENSIONS` flag for this PR, adding the extensions only when building using `build.mjs`
2. Remove the UMD build in a separate PR
3. Create an issue to use the package builds instead of `src` for the regression tests. In that issue's PR, we could revisit removing `MUI_ADD_IMPORT_EXTENSIONS` and modifying the regression infrastructure accordingly.

What do you think?

Reply:
Ok for me